mcp/src/main/java/io/modelcontextprotocol/server/McpAsyncServerExchange.java

```diff
@@ -101,4 +109,30 @@ public Mono<McpSchema.ListRootsResult> listRoots(String cursor) {
 				LIST_ROOTS_RESULT_TYPE_REF);
 	}
 
+	/**
+	 * Send a logging message notification to all connected clients. Messages below the
+	 * current minimum logging level will be filtered out.
+	 * @param loggingMessageNotification The logging message to send
+	 * @return A Mono that completes when the notification has been sent
+	 */
+	public Mono<Void> loggingNotification(LoggingMessageNotification loggingMessageNotification) {
+
+		if (loggingMessageNotification == null) {
+			return Mono.error(new McpError("Logging message must not be null"));
+		}
+
+		return Mono.defer(() -> {
+			if (loggingMessageNotification.level().level() < this.session.getMinLoggingLevel().level()) {
+				return Mono.empty();
+			}
+
+			return this.session.sendNotification(McpSchema.METHOD_NOTIFICATION_MESSAGE, loggingMessageNotification);
+		});
+	}
+
+	public void setMinLoggingLevel(LoggingLevel minLoggingLevel) {
```

Comment:
Public method requires a javadoc. Also, should this be exposed to the user? On one hand perhaps this should be in the user control, on the other it is an implementation of the spec if I'm correct, so maybe it should be hidden.

Reply:
I agree we can make it a protected so that only the server cha use it. We do not allow the users to register custom METHOD_LOGGING_SET_LEVEL. So this should remain spec implementation detail